test/json/component_status_list.json

```diff
@@ -0,0 +1,52 @@
+{
+  "kind": "ComponentStatusList",
+  "apiVersion": "v1",
+  "metadata": {
+    "selfLink": "/api/v1/componentstatuses"
+  },
+  "items": [
+    {
+      "metadata": {
+        "name": "controller-manager",
+        "selfLink": "/api/v1/namespaces/componentstatuses/controller-manager",
+        "creationTimestamp": null
+      },
+      "conditions": [
+        {
+          "type": "Healthy",
+          "status": "Unknown",
+          "error": "Get http://127.0.0.1:10252/healthz: dial tcp 127.0.0.1:10252: connection refused"
+        }
+      ]
+    },
+    {
+      "metadata": {
+        "name": "scheduler",
+        "selfLink": "/api/v1/namespaces/componentstatuses/scheduler",
+        "creationTimestamp": null
+      },
+      "conditions": [
+        {
+          "type": "Healthy",
+          "status": "Unknown",
+          "error": "Get http://127.0.0.1:10251/healthz: dial tcp 127.0.0.1:10251: connection refused"
+        }
+      ]
+    },
+    {
+      "metadata": {
+        "name": "etcd-0",
+        "selfLink": "/api/v1/namespaces/componentstatuses/etcd-0",
```

Comment:
As far as I understand, component statuses are resources/entities shared across the kubernetes cluster, not related to a specific namespace. 

one can list the component statuses in two ways (also according to swagger):
a. /api/v1/componentstatuses
b. /api/v1/namespaces/{namespace}/componentstatuses

or can get a specific component status: /api/v1/namespaces/{namespace}/componentstatuses/{name}

The selflink for a specific component status kubernetes currently provides in the response is broken. I will open an issue regarding it.